aries_cloudagent/connections/models/diddoc_v2/diddoc.py

```diff
@@ -239,11 +268,99 @@ def id(self, value: str) -> None:
 
         self._id = value
 
-    def set(
-        self,
-        item: Union[Service, VerificationMethod],
-        upsert=False,
-        verification_type="publicKey",
+    def add_verification_method(
+            self,
+            type: Union[PublicKeyType, str],
+            controller: Union[str, Sequence],
```

Comment:
I think controller should be an optional argument, defaulting to `self.id` when not given.